File: Sample/app/src/main/java/net/nend/sample/IconSpaceActivity.java
package net.nend.sample;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import net.nend.android.NendAdIconLoader;
import net.nend.android.NendAdIconView;

public class IconSpaceActivity extends AppCompatActivity {
	
	NendAdIconLoader mIconLoader;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.icon_space);
        
        mIconLoader = new NendAdIconLoader(getApplicationContext(), 101282, "0c734134519f25412ae9a9bff94783b81048ffbe");
        for ( int i = 1; i <= 4; i++ ) {
        	int id = getResources().getIdentifier("icon" + i, "id", getPackageName());
        	mIconLoader.addIconView((NendAdIconView)findViewById(id));
        }
        mIconLoader.loadAd();
    }

	@Override
	protected void onPause() {
		super.onPause();
		mIconLoader.pause();
	}

	@Override
	protected void onRestart() {
		super.onRestart();
		mIconLoader.resume();
	}    
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/icon/IconSpaceActivity.kt
package net.nend.sample.kotlin.icon

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import net.nend.android.NendAdIconLoader
import net.nend.android.NendAdIconView
import net.nend.sample.kotlin.R
import net.nend.sample.kotlin.icon.IconSampleActivity.Companion.ICON_API_KEY
import net.nend.sample.kotlin.icon.IconSampleActivity.Companion.ICON_SPOT_ID

class IconSpaceActivity : AppCompatActivity() {

    private lateinit var iconLoader: NendAdIconLoader

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.icon_space)

        iconLoader = NendAdIconLoader(applicationContext, ICON_SPOT_ID, ICON_API_KEY).apply {
            (1..4).map { i -> resources.getIdentifier("icon$i", "id", packageName) }
                    .forEach { addIconView(findViewById(it) as NendAdIconView) }
            loadAd()
        }
    }

    override fun onPause() {
        super.onPause()
        iconLoader.pause()
    }

    override fun onRestart() {
        super.onRestart()
        iconLoader.resume()
    }
}